### PR TITLE
Grid update

### PR DESCRIPTION
**Pull request by ararus**
Dropdown editor added
Tab moves the cursor to the first cell of the next row if the cursor is in the last column
New story to show how to implement server-side data mode
Cell range selection using keyboard
Paginated grid story added
Hotkeys added: Ctrl-A, Ctrl-Space (select column)

### CODE REVIEW

**Review by origami-z — COMMENTED** [review]
Some thoughts just on reading the [doc page](https://3f13f2df.uitk.pages.dev/?path=/docs/documentation-grid-grid--grid-variants):
- Three grids side by side feels a bit overwhelming. Maybe add some toggles (button/switch/etc.) to change the appearance rather than show them all at the same time.
- 2 props,`variant` and `zebra` toggling 3 variants can potentially introduce confusion in my mind. Maybe there is an opportunity to combine them? Discuss with designers as well.
- `cellSelectionMode` being `"none" | "range"` is a little bit confusing. Without reading the spec, i think it's `'single' | 'range' [| 'none']`?
- `columnDnD`, `DnD` may be not universally recognized prop naming. Something like `move` could be better?
- type of `defaultSelectedRowIdxs` being `Set`, will it cause issue around memorization as well as serialization for clients app?
- `onColumnMoved` as a `Grid` prop could be hard to work given the current args are 2 indices, it will be ok as a `Column` prop. On `Grid`, may need something to identify the Column on top of the indices, e.g. `id`.
- `onRowSelected`, can consider to align with List/Tree props. Also, with the current set up, it may have difficulty to implement grouping
- Under `Columns` section, could add an example with a few commonly used column configurations, so others can easily see the effect as well as the code in the doc
- `Row selection` section and example could be moved up to/around the `Grid` headings? Also would be benefiting from inline controls instead of side by side maybe?

File: packages/grid/src/internal/gridHooks.tsx [hunk 1]
```diff
@@ -1043,18 +1053,51 @@ export function useRangeSelection(cellSelectionMode?: GridCellSelectionMode) {
...
-        ref.current = {
+        mouseSelectionRef.current = {
           start: pos,
           unsubscribe: () => {
             document.removeEventListener("mouseup", onMouseUp);
             document.removeEventListener("mousemove", onMouseMove);
           },
         };
         setSelectedCellRange({ start: pos, end: pos });
+        9;
```
[the remark this answers is not in the document]
Comment (ararus) [reply packages/grid/src/internal/gridHooks.tsx]: Good spot!
Deleted

File: packages/grid/src/DropdownCellEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  flip,
+  limitShift,
+  shift,
+  size,
+} from "@floating-ui/react-dom-interactions";
```
[the remark this answers is not in the document]
Comment (ararus) [reply packages/grid/src/DropdownCellEditor.tsx]: Deleted

File: packages/grid/src/DropdownCellEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+export interface DropdownCellEditorProps<T> {
+  options: string[];
+  // Row and column are provided by the grid. See TableRow.tsx
+  row?: GridRowModel<T>;
+  column?: GridColumnModel<T>;
+}
+
+export function DropdownCellEditor<T>(props: DropdownCellEditorProps<T>) {
+  const { options, column, row } = props;
+  const [isOpen, setIsOpen] = useState<boolean>(true);
```
[the remark this answers is not in the document]
Comment (ararus) [reply packages/grid/src/DropdownCellEditor.tsx]: Done

**Review by EleanorHaproff — APPROVED** [review]

**Pull request merged**